app/controllers/application_controller.rb

```diff
@@ -4,14 +4,30 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :set_cache_header
 
-  helper_method :editable?
+  helper_method :editable?, :likeable?, :logged_in?, :current_developer
 
   private
 
+  def logged_in?
+    current_developer != nil
+  end
+
+  def current_developer
+    if session[:developer_id]
+      @current_developer ||= User.find(session[:developer_id])
```

Comment:
you can do:
```ruby
def current_developer
  @current_develope ||= User.find_by(email: session[:email])
end
```
it'd return nil if the email is not found, and I'm pretty sure omniauth gives you the email

Reply:
Updated also in the sessions_controller! 

`    session[:developer_email] = developer.email`